### PR TITLE
REP-3423 Migration-verifier should retry on Location28799 error while sampling documents

### PR DESCRIPTION
**Pull request by tdq45gj**
This PR adds `Location28799` error and make it retryable if running a pipeline with `$sample` stage.

### CODE REVIEW

**Review by autarch — APPROVED** [review]
LGTM!

File: internal/verifier/migration_verifier.go [hunk 1]
```diff
@@ -398,7 +398,7 @@ func (verifier *Verifier) getGenerationWhileLocked() (int, bool) {
 }
 
 func (verifier *Verifier) maybeAppendGlobalFilterToPredicates(predicates bson.A) bson.A {
-	if verifier.globalFilter == nil {
+	if len(verifier.globalFilter) == 0 {
```
Comment (FGasper) [inline internal/verifier/migration_verifier.go:401]: Why this change?

Comment (tdq45gj) [reply internal/verifier/migration_verifier.go:401]: This is to prevent an error like [REP-3390](https://jira.mongodb.org/browse/REP-3390). I don't think it's necessary here since we are only appending the filter to a slice of other predicates. But I think it makes slightly more sense to compare length with 0 than to check for a nil slice.

**Review by FGasper — APPROVED** [review]
LGTM. Thanks!

**Comment by tdq45gj** [comment]
evergreen merge 
This PR adds `Location28799` error and make it retryable if running a pipeline with `$sample` stage.
